dds/DCPS/Service_Participant.cpp

```diff
@@ -1923,6 +1925,9 @@ Service_Participant::load_domain_configuration(ACE_Configuration_Heap& cf,
         } else {
           reg->domain_default_config(domainId, tc);
         }
+      } else {
+        // store domain and associated config for template or multiple participant domain use
+        TheTransportRegistry->associate_domain_to_config(domainId, global_transport_config_.c_str());
```

Comment:
WChar builds are failing here since associate_domain_to_config is expecting std::string and global_transport_config_ is an ACE_TString (wrap with ACE_TEXT_ALWAYS_CHAR).